### ISSUE TITLE
String's max lengths should be not in bytes, but in characters

### ISSUE THREAD

**Issue by maximmax42** [Lachee/discord-rpc-csharp#218]
Consider this string of 128 Cyrillic letters а:
`аааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааа`
Counting in UTF-8 bytes, it's 256 bytes long, and yet, if I replace byte check in the library with a simple length comparison, it works just fine:
![image](https://user-images.githubusercontent.com/2225711/202428303-4fc9bb76-4d20-49d4-90d2-5e3d0b92ffbb.png)
Here's a string that has 64 Cyrillic а's and 64 Latin b's:
`ааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааааbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb`
Byte length would be 192, but character length is still 128, so it also works just fine:
![image](https://user-images.githubusercontent.com/2225711/202429053-dc11e15a-7cd5-44cf-9749-f7acc8098955.png)

**Comment by Lachee** [comment]
I'll look into this on my end. I will have to verify the API response because when this was first implemented it was byte count.

**Comment by maximmax42** [comment]
Yeah, it was probably changed, and it's not mentioned anywhere. Old page for this API still says 128 bytes, but it also says image names can be no more than 32 bytes, which we know isn't true anymore.

**Comment by maximmax42** [comment]
Oh and also, API doesn't accept text fields that are exactly 1 character, needs to be 2 or more.

**Comment by maximmax42** [comment]
An update: while button labels accept 32 characters and not bytes, putting a label with 32+ bytes still causes the whole presence to not show for others.

Discord Moment:tm:
